### PR TITLE
test: Name tests so they actually run

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Looks like there's a repeat of https://github.com/pola-rs/polars/issues/6316 😛 

Drive-by: use `MutablePlString` instead of `MutableBinaryViewArray::<str>`

### CODE REVIEW

File: py-polars/tests/unit/operations/test_sort.py [hunk 1]
```diff
@@ -48,10 +48,6 @@ def is_sorted_any(s: pl.Series) -> bool:
     return s.flags["SORTED_ASC"] or s.flags["SORTED_DESC"]
 
 
-def is_not_sorted(s: pl.Series) -> bool:
-    return not is_sorted_any(s)
```
Comment (MarcoGorelli) [inline py-polars/tests/unit/operations/test_sort.py:52]: another drive-by: removing this unused function 👻

**Review by ritchie46 — APPROVED** [review]
